File: 2023/1/1.cpp
#include <iostream>
#include <cstdio>
#include <vector>
#include <limits>
#include <algorithm>
#include <set>
#include <unordered_set>
using namespace std;

int check(int index, string a) {
    if (index + 5 <= a.length()) {
        if (a.substr(index, 5) == "three") return 3;
        if (a.substr(index, 5) == "seven") return 7;
        if (a.substr(index, 5) == "eight") return 8;
    }
    if (index + 4 <= a.length()) {
        if (a.substr(index, 4) == "four") return 4;
        if (a.substr(index, 4) == "five") return 5;
        if (a.substr(index, 4) == "nine") return 9;
    }
    if (index + 3 <= a.length()) {
        if (a.substr(index, 3) == "one") return 1;
        if (a.substr(index, 3) == "two") return 2;
        if (a.substr(index, 3) == "six") return 6;
    }
    return 0;
}

int main(){
    freopen("1.in", "r", stdin);
    int sum = 0;
    for (int i = 0; i < 1000; i++) {
        int first = 0;
        int second = 0;
        string a;
        cin >> a;
        for (int i = 0; i < a.length(); i++) {
            if(a[i] < 58 || (first = check(i, a))) {
                if (first == 0) first = a[i] - '0';
                break;
            }
        }
        for (int i = a.length()-1; i > 0; i--) {
            if(a[i] < 58 || (second = check(i-1, a))) {
                if (second == 0) second = a[i] - '0';
                break;
            }
        }
        if (second == 0) second = first;
        sum += (first*10 + second);
    }

    cout << sum;
}

File: 2023/1/1.py
pairs = [('one', 'o1e'), ('two', 't2'), ('three', 't3e'), ('four', '4'), ('five', '5e'), ('six', '6'), ('seven', '7n'), ('eight', 'e8t'), ('nine', 'n9e')]
s = 0
for line in open("1.in"):
    [(line := line.replace(w,d)) for w,d in pairs] #delete line for part 1
    l = [int(char) for char in line[0:-1] if ord(char) < 58 ]
    print(s := s + (l[0]*10 + l[-1]))

# print(sum := sum + (int(re.search(r'\d', line.rstrip()).group(0))*10 + int(re.search(r'\d', line.rstrip()[::-1]).group(0))))

#unreadable but 1 less line:
    # [(line := line.replace(w,d)) for w,d in pairs] #comment this line out for part 1
    # print(s := s + sum([10*int(k[0]) + int(k[-1]) for k in ["".join(filter(lambda x: ord(x)<58, line[0:-1]))]]))

#pt 1 one liner:
# print(sum([int(l[0] + l[-1]) for l in ["".join(filter(lambda x: x.isdigit(), line)) for line in open("1.in")]]))